xdsl/transforms/varith_transformations.py

```diff
@@ -64,6 +75,36 @@ def match_and_rewrite(self, op: Operation, rewriter: PatternRewriter, /):
                 return
 
 
+class VarithToArithPattern(RewritePattern):
+    """
+    Splits a varith operation into two arith operations.
+    """
+
+    def match_and_rewrite(self, op: Operation, rewriter: PatternRewriter, /):
+        # check that the op is of a type that we can convert to arith
+        if type(op) not in VARITH_TO_ARITH_TYPE_MAP:
+            return
+
+        # this must be true, as all keys of VARITH_TO_ARITH_TYPE_MAP are varith ops
+        op = cast(
+            varith.VarithOp,
+            op,
+        )
+
+        dest_type = VARITH_TO_ARITH_TYPE_MAP[type(op)]
```

Comment:
The map currently has duplicate keys, but you can distinguish between int/float by looking at `op.res.type` (try a debugger)